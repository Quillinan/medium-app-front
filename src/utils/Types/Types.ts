export interface AuthContextType {
  setAuth: (authToken: string | null) => void;
  removeAuth: () => void;
  token: string | null;
}

export interface ErrorResponse {
  code: string;
  message: string;
  detailedMessage: string;
  helpUrl: string;
  details: null;
}

export interface Birthday {
  NOME: string;
  DTNASCIMENTO: number;
}

export interface TableBirthdayProps {
  data: { NOME: string; DTNASCIMENTO: number }[];
}

export interface Post {
  id: number;
  title: string;
  subtitle: string;
  content: string;
  createdAt: Date;
  updatedAt: Date;
  authorId: string;
  authorName: string;
}
