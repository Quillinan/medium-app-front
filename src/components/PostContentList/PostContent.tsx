import { Post } from '@utils/Types/Types';

interface PostContentListProps {
  data: Post[];
}

const PostContentList: React.FC<PostContentListProps> = ({ data }) => {
  return (
    <div className='space-y-8'>
      {data.map(post => (
        <div
          key={post.id}
          className='flex flex-col md:flex-row border-b border-gray-200 pb-8'
        >
          <div className='flex-1 pr-4'>
            <h2 className='mt-1 text-2xl font-semibold text-gray-900'>
              {post.title}
            </h2>
            <p className='mt-2 text-gray-700'>{post.subtitle}</p>
            <p className='mt-1 text-sm text-gray-500'>
              Criado em:{' '}
              {new Date(post.createdAt).toLocaleDateString('pt-BR', {
                timeZone: 'UTC',
                year: 'numeric',
                month: '2-digit',
                day: '2-digit',
              })}{' '}
            </p>
            <p className='mt-1 text-sm text-gray-500'>
              Autor: {post.authorName}
            </p>
          </div>
          {post.coverImageUrl && (
            <div className='flex-shrink-0 mt-4 md:mt-0'>
              <img
                src={post.coverImageUrl}
                alt={post.title}
                className='h-40 object-cover'
              />
            </div>
          )}
        </div>
      ))}
    </div>
  );
};

export default PostContentList;
