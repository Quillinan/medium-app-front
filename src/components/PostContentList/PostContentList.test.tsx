import { render, screen } from '@testing-library/react';
import PostContentList from './PostContent';
import { Post } from '@utils/Types/Types';

describe('PostContentList', () => {
  const mockPosts: Post[] = [
    {
      id: 1,
      title: 'First Post',
      subtitle: 'Subtitle of First Post',
      content: 'This is the content of the first post.',
      createdAt: new Date('2024-10-29T00:00:00Z'),
      updatedAt: new Date('2024-10-29T00:00:00Z'),
      authorId: '123',
      authorName: 'Author One',
    },
    {
      id: 2,
      title: 'Second Post',
      subtitle: 'Subtitle of Second Post',
      content: 'This is the content of the second post.',
      createdAt: new Date('2024-10-28T00:00:00Z'),
      updatedAt: new Date('2024-10-28T00:00:00Z'),
      authorId: '456',
      authorName: 'Author Two',
    },
  ];

  it('should render nothing if there are no posts', () => {
    render(<PostContentList data={[]} />);

    expect(screen.queryByText('First Post')).not.toBeInTheDocument();
    expect(screen.queryByText('Second Post')).not.toBeInTheDocument();
  });

  it('should render the list of posts', () => {
    render(<PostContentList data={mockPosts} />);

    expect(screen.getByText('First Post')).toBeInTheDocument();
    expect(screen.getByText('Subtitle of First Post')).toBeInTheDocument();
    expect(screen.getByText('Criado em: 29/10/2024')).toBeInTheDocument();
    expect(screen.getByText('Autor: Author One')).toBeInTheDocument();

    expect(screen.getByText('Second Post')).toBeInTheDocument();
    expect(screen.getByText('Subtitle of Second Post')).toBeInTheDocument();
    expect(screen.getByText('Criado em: 28/10/2024')).toBeInTheDocument();
    expect(screen.getByText('Autor: Author Two')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { asFragment } = render(<PostContentList data={mockPosts} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
