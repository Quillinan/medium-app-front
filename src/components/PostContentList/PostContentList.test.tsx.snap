// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PostContentList > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="space-y-8"
  >
    <div
      class="flex flex-col border-b border-gray-200 pb-8"
    >
      <h2
        class="mt-1 text-2xl font-semibold text-gray-900"
      >
        First Post
      </h2>
      <p
        class="mt-2 text-gray-700"
      >
        Subtitle of First Post
      </p>
      <p
        class="mt-1 text-sm text-gray-500"
      >
        Criado em: 29/10/2024 
      </p>
      <p
        class="mt-1 text-sm text-gray-500"
      >
        Autor: Author One
      </p>
    </div>
    <div
      class="flex flex-col border-b border-gray-200 pb-8"
    >
      <h2
        class="mt-1 text-2xl font-semibold text-gray-900"
      >
        Second Post
      </h2>
      <p
        class="mt-2 text-gray-700"
      >
        Subtitle of Second Post
      </p>
      <p
        class="mt-1 text-sm text-gray-500"
      >
        Criado em: 28/10/2024 
      </p>
      <p
        class="mt-1 text-sm text-gray-500"
      >
        Autor: Author Two
      </p>
    </div>
  </div>
</DocumentFragment>
`;
